report/main.tex

```diff
@@ -124,7 +124,7 @@ \subsubsection{CNN for detecting non-runnable layer}
 
 \subparagraph{Creating training and validation sets with masks\\}
 
-In order to train the CNN with a limited data-set, we took maps without extra layers, in other words maps that were ready for path detection, and added layers on top of them so that we could have ample training and validation sets. This included drawing a random number of circles, lines, and text using cv2 functions, ensuring that the locations of the circles were spread out and the lines connected the circles, and using a variety of colors (though uniform per training image). All in all it was fairly simple and allowed us to produce however many training/validation images as we wanted. To produce the masks for the training and validation sets, all that needed to be done was feed in the color that was used to create the figures on the training/validation images into a function that took a “bitwise and” of the image and a mask of that color. Then just converting the mask to black and white produced good results.
+In order to train the CNN with a limited data-set, we took maps without extra layers, in other words maps that were ready for path detection, and added layers on top of them so that we could have ample training and validation sets. This included drawing a random number of circles, lines, and text using OpenCV functions, ensuring that the locations of the circles were spread out and the lines connected the circles, and using a variety of colors (though uniform per training image). The circle drawing amounted to selecting a random position inside the bounds of the image, and checking other circles to ensure that no circles were overlapping, given a set radius. Drawing the lines became about connecting the Nth circle with the (N-1)st circle, which with some basic trigonometry was done. Finally labels were added on the image for each circle, and placed in the vicinity of the circles. OpenCV gave us the tools to draw the figures on the image, so we had a function that would essentially take in an image and output that image with a set amount of lines/circles/labels of some defined radius, line width, color, and number. All in all it was fairly simple and allowed us to produce however many training/validation images as we wanted. To produce the masks for the training and validation sets, all that needed to be done was feed in the color that was used to create the figures on the training/validation images into a function that took a “bitwise and” of the image and a mask of that color. In other words every pixel that was of the same color as the lines/circles was kept in the mask, and those that were not were replaced with black. This gave us a pink mask, which was then converted to black and white.
```

Comment:
```suggestion
In order to train the CNN with a limited data-set, we took maps without circuit layer, in other words maps that were ready for path detection, and added layers on top of them so that we could have ample training and validation sets. This included drawing a random number of circles, lines, and text using OpenCV functions, ensuring that the locations of the circles were spread out and the lines connected the circles, and using a variety of colors (though uniform per training image). The circle drawing amounted to selecting a random position inside the bounds of the image, and checking other circles to ensure that no circles were overlapping, given a set radius. Drawing the lines became about connecting the Nth circle with the (N-1)st circle, which with some basic trigonometry was done. Finally labels were added on the image for each circle, and placed in the vicinity of the circles. OpenCV gave us the tools to draw the figures on the image, so we had a function that would essentially take in an image and output that image with a set amount of lines/circles/labels of some defined radius, line width, color, and number. All in all it was fairly simple and allowed us to produce however many training/validation images as we wanted. To produce the masks for the training and validation sets, all that needed to be done was feed in the color that was used to create the figures on the training/validation images into a function that took a “bitwise and” of the image and a mask of that color. In other words every pixel that was of the same color as the lines/circles was kept in the mask, and those that were not were replaced with black. This gave us a pink mask, which was then converted to black and white.
```